### PR TITLE
CONSOLE-4969: Configure a coderabbit.yaml file for console repo

### PR DESCRIPTION
**Pull request by krishagarwal278**
## Summary by CodeRabbit

* **Chores**
  * Added automated code review configuration to establish review standards and documentation references for the repository.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: .ai/coderabbit/.coderabbit.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# Configuration Philosophy: Senior Engineer Review Standards
+# This config is tuned to provide reviews that match senior software engineer expectations:
+# - Architecture & Design: Focus on system design, patterns, and scalability
+# - Security: Identify security vulnerabilities and best practices
+# - Performance: Flag performance issues and optimization opportunities
+# - Maintainability: Ensure code is readable, testable, and maintainable
+# - Breaking Changes: Critical attention to public APIs and backward compatibility
+# - Testing: Comprehensive test coverage and edge case handling
+# - Documentation: Clear code documentation and API contracts
+# - Context Awareness: Understanding of project architecture and conventions
```
Comment (sg00dwin) [inline .ai/coderabbit/.coderabbit.yaml]: This should be provided in the `tone_instructions` value. Adhering to coderabbit max character limit 250

I managed to condense most of it right to 250 characters.

```suggestion
tone_instructions: "Act as a principal engineer reviewing OpenShift Console (React/TS/JS/Go). Focus: architecture, security (RBAC/SCC), performance, maintainability, backward-compatible APIs, thorough testing/edge cases. Enforce K8s best practices, PatternFly, a11y/i18n, efficient resource watching"
```

File: .coderabbit.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+# This config is tuned to provide reviews that match senior software engineer expectations:
+# - Architecture & Design: Focus on system design, patterns, and scalability
+# - Security: Identify security vulnerabilities and best practices
+# - Performance: Flag performance issues and optimization opportunities
+# - Maintainability: Ensure code is readable, testable, and maintainable
+# - Breaking Changes: Critical attention to public APIs and backward compatibility
+# - Testing: Comprehensive test coverage and edge case handling
+# - Documentation: Clear code documentation and API contracts
+# - Context Awareness: Understanding of project architecture and conventions
+
```
Comment (sg00dwin) [inline .coderabbit.yaml:2]:
```suggestion
early_access: false
```

File: .ai/coderabbit/.coderabbit.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+reviews:
+  # Disable fortune messages for professional reviews
+  in_progress_fortune: false
+
+  # Assertive profile for thorough, senior-engineer-level reviews
+  # This ensures comprehensive feedback on architecture, security, performance, and best practices
+  profile: assertive
+
+  # Request changes workflow to ensure quality standards
+  request_changes_workflow: true
```
Comment (sg00dwin) [inline .ai/coderabbit/.coderabbit.yaml]: The default setting is `false`
true implies `Approve the review once CodeRabbit's comments are resolved`

I don't think that is what we want.

File: .ai/coderabbit/.coderabbit.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  # Show review status for transparency
+  review_status: true
+
+  # Auto-review configuration
+  auto_review:
+    enabled: true
+    # Review drafts to catch issues early
+    drafts: true
+
```
Comment (sg00dwin) [inline .ai/coderabbit/.coderabbit.yaml]:
```suggestion
ignore_title_keywords:
      - WIP
      - '[WIP]'
      - do-not-merge
      - work-in-progress
```

File: .ai/coderabbit/.coderabbit.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
```
Comment (sg00dwin) [inline .ai/coderabbit/.coderabbit.yaml]:
```suggestion
labels:
      - '!do-not-merge/work-in-progress'
```

File: .ai/coderabbit/.coderabbit.yaml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  fail_commit_status: false
+
+  # Don't collapse walkthrough for better visibility
+  collapse_walkthrough: false
+
+  # Show summary of changed files
+  changed_files_summary: true
+
+  # Generate sequence diagrams for complex flows
+  sequence_diagrams: true
```
Comment (sg00dwin) [inline .ai/coderabbit/.coderabbit.yaml]: These can be requested by a reviewer, on a case by case basis, in the pr comments via 
`@coderabbitai generate sequence diagram`

```suggestion
  sequence_diagrams: false
```

File: .ai/coderabbit/.coderabbit.yaml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+# - Documentation: Clear code documentation and API contracts
+# - Context Awareness: Understanding of project architecture and conventions
+
+reviews:
+  # Disable fortune messages for professional reviews
+  in_progress_fortune: false
+
+  # Assertive profile for thorough, senior-engineer-level reviews
+  # This ensures comprehensive feedback on architecture, security, performance, and best practices
+  profile: assertive
```
Comment (sg00dwin) [inline .ai/coderabbit/.coderabbit.yaml]: I think we want to leave this with at its default value. [There own docs](https://docs.coderabbit.ai/reference/configuration#param-profile) say `assertive` "yields more feedback, that may be considered nitpicky." Which might be overloaded.

```suggestion
  profile: chill
```

File: .ai/coderabbit/.coderabbit.yaml (hunk 1 above)
Comment (krishagarwal278) [reply .ai/coderabbit/.coderabbit.yaml]: The instruction is exceeding total character count so keeping it as:
`tone_instructions: "Principal engineer reviewing OpenShift Console (React/TS/JS/Go). Focus: architecture, security (RBAC/SCC), performance, maintainability, backward-compatible APIs, testing/edge cases. Enforce K8s best practices, PatternFly, a11y/i18n."`

File: .ai/coderabbit/.coderabbit.yaml (hunk 3 above)
Comment (krishagarwal278) [reply .ai/coderabbit/.coderabbit.yaml]: so we do want it to be less enforced. gotcha 👍🏻 😄

**Comment by sg00dwin** [comment]
Also, this file should reside at the root of console.
